### PR TITLE
update/group

### PR DESCRIPTION
**Pull request by till**
- Chore: add a --debug flag and output something
- Update: add group variable to each host

Related: pngmbh/issues#980

### CODE REVIEW

File: inventory/builder.go [hunk 1]
```diff
@@ -3,23 +3,17 @@ package inventory
...
-		if node.FloatingIP != "" {
-			inventory.AddHostVar("floating_ip", node.FloatingIP, node.Name)
-		}
-
-		if len(node.MetaData) > 0 {
-			node_swarm_label, ok := node.MetaData["com.planetary-quantum.meta.label"]
-			if ok {
-				inventory.AddHostVar("swarm_labels", node_swarm_label, node.Name)
-			}
+		hostVars := host.Build(node)
```
Comment (till) [inline inventory/builder.go:14]: I think this is a nice accidental find. Basically use the same method for `--list` and individual queries via `--host=foo`.

File: main.go [hunk 2]
```diff
@@ -41,6 +43,12 @@ func main() {
 			},
 		},
 		Flags: []cli.Flag{
+			&cli.BoolFlag{
```
Comment (till) [inline main.go:46]: Figured might as well. It's not set when Ansible invokes it, but when "shit goes wrong" we have a way to add `--debug` cause then it's interactive anyway. ;)

File: host/host.go [hunk 3]
```diff
@@ -12,11 +14,21 @@ func Build(node server.AnsibleServer) map[string]string {
 		hostVars["floating_ip"] = node.FloatingIP
 	}
 
-	if len(node.MetaData) > 0 {
-		node_swarm_label, ok := node.MetaData["com.planetary-quantum.meta.label"]
-		if ok {
-			hostVars["swarm_labels"] = node_swarm_label
-		}
+	if len(node.MetaData) == 0 {
```
Comment (till) [inline host/host.go:17]: Re-arranged this to "exit early".

File: host/host.go [hunk 4]
```diff
@@ -12,11 +14,21 @@ func Build(node server.AnsibleServer) map[string]string {
...
+		log.Debugf("Empty/broken metadata for %s", node.Name)
+		return hostVars
+	}
+
+	node_swarm_label, ok := node.MetaData["com.planetary-quantum.meta.label"]
+	if ok {
+		hostVars["swarm_labels"] = node_swarm_label
+	}
+
+	node_group, err := server.GetGroup(node)
```
Comment (till) [inline host/host.go:27]: This part is new.

File: server/metadata.go [hunk 5]
```diff
@@ -1,6 +1,22 @@
 package server
 
+import "fmt"
+
 const role_label string = "com.planetary-quantum.meta.role"
+const group_label string = "com.planetary-quantum.meta.customer_group"
+
+func GetGroup(server AnsibleServer) (string, error) {
```
Comment (till) [inline server/metadata.go:8]: This is where I extract the name from the instance metadata.

Comment (till) [reply server/metadata.go:8]: Not really sure if I use the constants correctly. Maybe this should be something else. Basically looking to declare a type that I can consume in other apps as well. But no idea, this also works.

Comment (rwos) [reply server/metadata.go:8]: yeah I think this is okay

**Review by rwos — APPROVED** [review]
haven't tried it out but LGTM

**Pull request merged**